handle SSLOption.

n0rmanc opened the pull request:
In the Azure CosmosDB Cassandra. We need to setup sslMode require. 
Reference from quick start guide of Azure cosmos db.
`cqlsh service.cassandra.cosmos.azure.com 10350 -u userName -p password --ssl`

n0rmanc committed:
handle SSLOption.

gravis commented on cassandra.go:
```diff
@@ -75,6 +75,20 @@ func Open(rawurl string) (driver.Driver, error) {
 
 	}
+	// handle ssl option
+	if len(u.Query().Get("sslmode")) > 0 {
```
For the sake of code consistency, shouldn't we have 

```go
if sslMode, ok := u.Query()["sslmode"]; ok {
```

instead? If this syntax is more idiomatic, then would you mind changing the rest of the file accordingly?
The next line would not be needed then.

gravis replied:
This code doesn't make sense to me: 

```
if sslmode := u.Query().Get("sslmode"); sslmode != "" {
	if sslmode != "disable" {
```

Why don't you use instead: `if sslmode := u.Query().Get("sslmode"); sslmode != "disable" {``

n0rmanc replied:
`u.Query().Get("sslmode")` could return zero-value if we don't assign `sslmode`.
It will enter the block and init `cluster.SslOpts`.
how about :
```
if sslmode := u.Query().Get("sslmode"); sslmode != "" && sslmode != "disable" {
```

gravis replied:
ha ok. In this case, shouldn't it be `sslmode != "" || sslmode != "disable" instead?
Because `sslmode != "" && sslmode != "disable"` is pretty much only `sslmode != "disable"`

n0rmanc replied:
`sslmode != "" || sslmode != "disable"`, In this condition, if will init `cluster.SslOpts`, if the user doesn't set `sslmode`.
But I think, we shouldn't init `cluster.SslOpts`.
Because of the user doesn't specific `sslmodel` in the URL.

The following case we should init `cluster.SslOpts`.
1. user set `sslmode`, this mean u.Query().Get("sslmode") is not empty string. 
condition: `sslmode != ""`.
2. `sslmode` is not disable.
condition: `sslmode != "disable"`

Combine 1. and 2., the condition will be `sslmode!= "" && sslmode != "disable"`.

gravis replied:
Ok, thanks for sharing. I get your point.

gravis commented on cassandra.go:
```diff
@@ -75,6 +75,20 @@ func Open(rawurl string) (driver.Driver, error) {
+	if len(u.Query().Get("sslmode")) > 0 {
+		if u.Query().Get("sslmode") != "disable" {
+			cluster.SslOpts = &gocql.SslOptions{}
+			if len(u.Query().Get("sslrootcert")) > 0 && len(u.Query().Get("sslcert")) > 0 && len(u.Query().Get("sslkey")) > 0 {
```
At this point (`sslmode` has been tested), I think it's safe to assign these values directly.
They will default to zero values anyway. What do you think?

gravis reviewed (commented):
Thanks for your contribution! 
Could you please check my few comments?

n0rmanc committed:
Address comments.

String zero value is empty string. Query().Get() return empty string, too. Assign value directly without checkt.

n0rmanc committed:
Refactors len(u.Query().Get("sslmode")) > 0

Get sslmode in the if statements and reuse it.

n0rmanc committed:
move the condition to same line

n0rmanc commented:
@gravis when will `journey` get update?
Should I create a PR for it?

gravis commented:
I've tried to release a new version, but migrate is failing on recent go versions, because of https://github.com/golang/go/issues/33646

We're using `net/url` to [parse the DSN](https://github.com/db-journey/migrate/blob/master/driver/driver.go#L79), and it's not working anymore. Some rewrite work is required :(
